Geometry/MTDNumberingBuilder/src/MTDTopology.cc

```diff
@@ -1,3 +1,134 @@
 #include "Geometry/MTDNumberingBuilder/interface/MTDTopology.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
 
-MTDTopology::MTDTopology(const int &topologyMode) : mtdTopologyMode_(topologyMode) {}
+MTDTopology::MTDTopology(const int& topologyMode, const ETLValues& etl)
+    : mtdTopologyMode_(topologyMode), etlVals_(etl) {}
+
+bool MTDTopology::orderETLSector(const GeomDet*& gd1, const GeomDet*& gd2) {
+  ETLDetId det1(gd1->geographicalId().rawId());
+  ETLDetId det2(gd2->geographicalId().rawId());
+
+  if (det1.mtdRR() != det2.mtdRR()) {
+    return det1.mtdRR() < det2.mtdRR();
+  } else if (det1.modType() != det2.modType()) {
+    return det1.modType() < det2.modType();
+  } else {
+    return det1.module() < det2.module();
+  }
+}
+
+size_t MTDTopology::hshiftETL(const uint32_t detid, const int horizontalShift) const {
+  size_t failIndex(std::numeric_limits<unsigned int>::max());  // return out-of-range value for any failure
```

Comment:
Shouldn't `failIndex` be `static const`?

Reply:
```const``` for sure, but also ```static``` does not harm I would say